test/language/expressions/yield/star-iterable.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 25.2
+description: >
+    When an iterator is the operand of a `yield *` expression, the generator
+    should produce an iterator that visits each iterated item.
+---*/
+
+var results = [{ value: 1 }, 13, 'foo', { value: 34, done: true }];
```

Comment:
Right you are. I think that behavior warrants a dedicated test, so I think I will simplify this to return a valid object for each iteration.